src/bci_build/package/__init__.py

```diff
@@ -1428,15 +1502,15 @@ def __post_init__(self) -> None:
 
     @property
     def _registry_prefix(self) -> str:
-        if self.os_version.is_tumbleweed:
-            return "opensuse"
-        return "suse"
+        assert self.registry_type
+        return self.registry_type.application_prefix
 
     @property
     def image_type(self) -> ImageType:
         return ImageType.APPLICATION
 
     @property
+    @Registry.app_coll_title
```

Comment:
I don't think such a mutating decorator makes understanding the code logic any easier. 

what about simply delegating that to a method in a RegistryClass (see the `def url(self, container)`) example in #1443  instead?